ert_storage/endpoints/records.py

```diff
@@ -582,49 +582,62 @@ def _get_and_assert_ensemble(
 async def _get_record_data(record: ds.Record, accept: Optional[str]) -> Response:
     type_ = record.record_type
     if type_ == ds.RecordType.float_vector:
-        if accept == "application/x-numpy":
-            from numpy.lib.format import write_array
-
-            stream = io.BytesIO()
-            write_array(stream, np.array(record.f64_matrix.content))
-
-            return Response(
-                content=stream.getvalue(),
-                media_type="application/x-numpy",
-            )
-        if accept == "application/x-dataframe":
-            data = pd.DataFrame(record.f64_matrix.content)
-            labels = record.f64_matrix.labels
-            if labels is not None:
-                data.columns = labels[0]
-                data.index = labels[1]
-            return Response(
-                content=data.to_csv().encode(),
-                media_type="application/x-dataframe",
-            )
-        else:
-            return record.f64_matrix.content
+        return _extract_float_vector(record=record, accept=accept)
```

Comment:
Not sure if numpy stays in the future, but if it will I would split this like `_extract_numpy` and `_extract_pandas`?

Reply:
I now do the `if` on `RecordType` - which is of kind `float_vector` or `file`. If I for the `float_vector` should do another `if` based on the `accept` and choose between `_extract_numpy` and `_extract_pandas` - then why should we refactor at all? I guess the purpose of the refactoring is to separate away part of the if's and make it more clean. 

That said, you might be correct - but from my point of view i want to have the different paths separated into its own functions.

Reply:
I see, just wanted to have something more general (eg. Response for numpy, file, pandas). Maybe this suffices for now and we see if there is a need for it later on.